src/iexit.cpp

```diff
@@ -24,14 +26,70 @@ main( int argc, char **argv ) {
     }
 
     rodsEnv myEnv;
-    getRodsEnv( &myEnv );
-    char* envFile = getRodsEnvFileName();
-    status = unlink( envFile );
-    if ( myRodsArgs.verbose == True ) {
-        printf( "Deleting (if it exists) session envFile: %s\n", envFile );
-        printf( "unlink status [%d]\n", status );
-        // prompt before removing
-        obfRmPw( 0 );
+    getRodsEnv(&myEnv);
+    const char* envFile = getRodsEnvFileName();
+    bool isServiceAccountError = false;
+    bool isServiceAccount = false;
+    try {
+        isServiceAccount = irods::is_service_account();
+    }
+    catch (const irods::exception &e) {
+        isServiceAccountError = true;
+        isServiceAccount = true; // to make if/else blocks neater
+        std::fprintf(stderr,
+                     "WARNING: Could not determine whether %s is running as service account "
+                     "due to thrown exception: %s",
+                     argv[0],
+                     e.client_display_what());
+    }
+
+    if (myRodsArgs.verbose == True) {
+        std::printf("Deleting (if it exists) session envFile: %s\n", envFile);
+    }
+    status = unlink(envFile);
+    if (myRodsArgs.verbose == True) {
+        std::printf("unlink status [%d]\n", status);
+    }
+
+    if (myRodsArgs.verbose == True && !isServiceAccount) {
+        std::printf("Deleting (if it exists) session auth file\n");
+    }
+    if (myRodsArgs.force == True) {
+        if (isServiceAccount) {
+            // MAYBE TODO: try to determine if auth file is safe to remove due to env variables
```

Comment:
Leftover? Perhaps this should be captured in an issue.

Reply:
#6986